tests/test_groups.py

```diff
@@ -107,3 +108,15 @@ def test_fully_connect_groups_equivalence(SimpleNet):
     fully_connect(pre, post, IonotropicSynapse())
 
     assert all(net1.edges == net2.edges)
+
+
+def test_set_ncomp_changes_groups(SimpleCell):
+    """Test whether groups get updated appropriately after `set_ncomp`."""
+    cell = SimpleCell(3, 4)
+    cell.branch(0).add_to_group("exc")
+    cell.branch(0).set_ncomp(1)
+    assert len(cell.exc.nodes) == 1
+
+    cell.branch(1).add_to_group("exc")
+    cell.branch(0).set_ncomp(2)
+    assert len(cell.exc.nodes) == 6  # 2 from branch(0) and 4 from branch(1).
```

Comment:
also might be worth testing if `set_ncomp` correctly raises if ngroups > 1 within module.